api/institutions/views.py

```diff
@@ -0,0 +1,247 @@
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.INSTITUTION_READ, CoreScopes.NODE_BASE_READ]
+    required_write_scopes = [CoreScopes.NULL]
+    model_class = Node
+
+    serializer_class = NodeSerializer
+    view_category = 'institutions'
+    view_name = 'institution-nodes'
+
+    # overrides ODMFilterMixin
+    def get_default_odm_query(self):
+        inst = self.get_institution()
+        inst_query = Q('primary_institution', 'eq', inst)
+        base_query = (
+            Q('is_deleted', 'ne', True) &
+            Q('is_folder', 'ne', True) &
+            Q('is_registration', 'eq', False)
+        )
+        user = self.request.user
+        permission_query = Q('is_public', 'eq', True)
+        if not user.is_anonymous():
+            permission_query = (permission_query | Q('contributors', 'icontains', user._id))
+
+        query = base_query & permission_query & inst_query
+        return query
+
+    # overrides RetrieveAPIView
+    def get_queryset(self):
+        query = self.get_query_from_request()
+        return Node.find(query)
+
+
+class InstitutionNodeDetail(JSONAPIBaseView, generics.RetrieveAPIView, NodeMixin, InstitutionMixin):
+    """Detail of a node with this primary institution
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        ContributorOrPublic
+    )
+
+    required_read_scopes = [CoreScopes.INSTITUTION_READ, CoreScopes.NODE_BASE_READ]
+    required_write_scopes = [CoreScopes.NULL]
+    model_class = Node
+
+    serializer_class = NodeDetailSerializer
+    view_category = 'institutions'
+    view_name = 'institution-node-detail'
+
+    # overrides RetrieveAPIView
+    def get_object(self):
+        inst = self.get_institution()
+        node = self.get_node()
+        if node.primary_institution != inst:
+            raise NotFound
+        return node
+
+
+class InstitutionUserList(JSONAPIBaseView, ODMFilterMixin, generics.ListAPIView, InstitutionMixin):
+    """Users that have been authenticated with the institution.
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.INSTITUTION_READ, CoreScopes.USERS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+    model_class = User
+
+    serializer_class = UserSerializer
+    view_category = 'institutions'
+    view_name = 'institution-users'
+
+    # overrides ODMFilterMixin
+    def get_default_odm_query(self):
+        inst = self.get_institution()
+        query = Q('affiliated_institutions', 'eq', inst)
+        return query
+
+    # overrides RetrieveAPIView
+    def get_queryset(self):
+        query = self.get_query_from_request()
+        return User.find(query)
+
+
+class InstitutionUserDetail(JSONAPIBaseView, generics.RetrieveAPIView, InstitutionMixin):
```

Comment:
Related to this comment: https://github.com/CenterForOpenScience/osf.io/pull/4661/files#r47368430

Is this a needed view?